### DISCUSSION TITLE
Documentation about EXTENSION_NAME_REGEX?

### DISCUSSION THREAD

**Discussion by CapitaineToinon** [directus/directus#14417] [Archive]
I was curious to know how interfaces managed to load automatically after being installed in `node_modules` and stumbled upon `EXTENSION_NAME_REGEX`. This method is super useful in a monorepo setup where I have my frontend in one package, directus in a 2nd package and finally my extension in a third and as a dependency of my directus package.

My question is, since I couldn't find any documentation about this, is it recommended to rely on this method? And if yes, could you add a section about it in [Deploying your extension](https://docs.directus.io/extensions/creating-extensions/#deploying-your-extension)?

Thanks 🙏

**Comment by rijkvanzanten** [comment]
Yes, it is safe to rely on this method of loading extensions 👍🏻 Like you correctly stated, we're using it to automatically discover extensions installed through `npm`.

**Reply by rijkvanzanten** [reply]
(cc @nickrum / @erondpowell for docs)

**Reply by erondpowell** [reply]
logged in issues
